apps/speed.c

```diff
@@ -1066,6 +1094,24 @@ static int RSA_verify_loop(void *args)
 }
 #endif
 
+#ifndef OPENSSL_NO_DH
+static long ffdh_c[FFDH_NUM][1];
+
+static int FFDH_derive_key_loop(void *args)
+{
+        loopargs_t *tempargs = *(loopargs_t **) args;
+        DH *privkey = tempargs->ffdh_ctx[testnum];
+        BIGNUM *pubkey = tempargs->ffdh_pubkey[testnum];
+        unsigned char *derived_secret = tempargs->secret_ff_a;
+        int count;
+
+        for (count = 0; COND(ffdh_c[testnum][0]); count++)
+            DH_compute_key(derived_secret, pubkey, privkey);
```

Comment:
Should we be adding *new* uses of deprecated APIs? speed is kind-of special, but the idea is that all the other deprecated uses will be removed from speed as and when the deprecated APIs are also removed.

Reply:
I was looking for way how to do it with the EVP API, but didn't found definitions for the FFDHE parameters from the RFC.

Reply:
@mattcaswell did I miss that documentation or it's not there?

Reply:
You can still use `DH_new_by_nid()` (that is *not* deprecated). You can then use `EVP_PKEY_assign_DH` to create an EVP_PKEY from it. Then you can use EVP_PKEY_derive* to derive a shared key.

Reply:
I wasn't able to make `EVP_PKEY_assign_DH` work with current master, I've used `EVP_PKEY_CTX_new_id(EVP_PKEY_DH, NULL)` in the end

Reply:
> I wasn't able to make EVP_PKEY_assign_DH work with current master, I've used EVP_PKEY_CTX_new_id(EVP_PKEY_DH, NULL) in the end

Hmm. Ok...although I don't understand why `EVP_PKEY_assign_DH` did not work...it *should* do

Reply:
@mattcaswell IIRC without calling `EVP_PKEY_CTX_set_dh_nid` the `EVP_PKEY_keygen` complained about missing parameters in context and when both `EVP_PKEY_assign_DH` and `EVP_PKEY_CTX_set_dh_nid` were used the `EVP_PKEY_keygen` failed without logging errors (`ERR_print_errors` didn't print anything)